http3/multipart.py

```diff
@@ -15,9 +15,13 @@ def render_data(self) -> bytes:
 
 
 class DataField(Field):
-    def __init__(self, name: str, value: str) -> None:
+    def __init__(self, name: str, value: typing.Union[str, bytes]) -> None:
         self.name = name
         self.value = value
+        if not isinstance(self.name, str):
+            raise TypeError("Invalid type for name. Expected str.")
+        if not isinstance(self.value, (str, bytes)):
+            raise TypeError("Invalid type for value. Expected str or bytes.")
```

Comment:
Is there a more elegant way to write this 4 lines above?

Reply:
Seems fine to me, maybe if we end up doing this a lot we'd want a function that checks the type of a value and raises a TypeError with the given message if it's not valid?

Reply:
Such function would be eventually useful. Type hinting could be optionally stricter in sense that could raise runtime errors.